### PR TITLE
added mocking of transaction submission for easier development

### PR DESCRIPTION
**Pull request by refi93**
usage explained in README.

- check that the explanation is clear.
- check that tx submission mocking actually works as described

### CODE REVIEW

File: server/transactionSubmitter.js [hunk 1]
```diff
@@ -24,7 +25,15 @@ module.exports = function(app, env) {
       return res.status(500).send('bad request format')
     }
 
-    // return res.end(JSON.stringify({success: true, txHash}))
+    if (process.env.CARDANOLITE_MOCK_TX_SUBMISSION === 'true') {
+      await sleep(5000)
+      return res.end(
+        JSON.stringify({
+          success: process.env.CARDANOLITE_MOCK_TX_SUBMISSION_RESPONSE === 'true',
```
Comment (git-jano) [inline server/transactionSubmitter.js]: shouldn't it be CARDANOLITE_MOCK_TX_SUBMISSION_SUCCESS ?

File: server/blockchainExplorerMock.js [hunk 2]
```diff
@@ -16,7 +21,7 @@ module.exports = function(app, env) {
     const txId = req.params.txId
 
     const response = await request(
-      `${process.env.CARDANOLITE_BLOCKCHAIN_EXPLORER_PROXY_TARGET}/api/txs/raw/${txId}`
+      `${process.env.CARDANOLITE_MOCK_BLOCKCHAIN_EXPLORER_TARGET}/api/txs/raw/${txId}`
```
Comment (git-jano) [inline server/blockchainExplorerMock.js]: misleading naming if the target is not mocked

File: server/helpers/loadFrontendConfig.js [hunk 3]
```diff
@@ -1,8 +1,17 @@
 // config intended to be passed on to frontend
 // keys in process .env must be referencesd explicitly
 // so dotenv-webpack is able to include them in the test bundle for Mocha tests
+
+function getBlockchainExplorerUrlForFrontend() {
+  if (process.env.CARDANOLITE_ENABLE_MOCK_SERVER === 'true') {
+    return `http://localhost:${process.env.PORT}`
+  } else {
```
Comment (git-jano) [inline server/helpers/loadFrontendConfig.js]: what about ternary operator? is it short enough to use it here?

File: app/frontend/helpers/debugLog.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const CARDANOLITE_CONFIG = require('../config').CARDANOLITE_CONFIG
 
 function debugLog(message) {
-  if (CARDANOLITE_CONFIG.CARDANOLITE_ENABLE_DEBUGGING) {
+  if (CARDANOLITE_CONFIG && CARDANOLITE_CONFIG.CARDANOLITE_ENABLE_DEBUGGING) {
```
Comment (git-jano) [inline app/frontend/helpers/debugLog.js]: is this useful? who would bebug without config?

File: app/frontend/wallet/cardano-wallet.js [hunk 5]
```diff
@@ -62,13 +62,13 @@ const CardanoWallet = async (options) => {
...
       throw NamedError('TransactionRejectedByTrezor')
     })
+
     const response = await blockchainExplorer
       .submitTxRaw(signedTx.txHash, signedTx.txBody)
       .catch((e) => {
         debugLog(e)
         throw NamedError('TransactionRejectedByNetwork')
       })
-    updateUtxosFromTxAux(txAux)
```
Comment (git-jano) [inline app/frontend/wallet/cardano-wallet.js]: how is this replaced?

Comment (refi93) [reply app/frontend/wallet/cardano-wallet.js]: thanks for pointing out, I forgot to reverse this change after debugging the tests

File: app/frontend/helpers/debugLog.js (hunk 4 above)
Comment (refi93) [reply app/frontend/helpers/debugLog.js]: the tests fail without this. There is already an issue to refactor debug logging: https://github.com/vacuumlabs/cardanolite/issues/161

Comment (refi93) [reply app/frontend/helpers/debugLog.js]: but it fails only if something is logged which isn't if everything goes as expected. Ok, dropping this change for now.

**Pull request merged**